backend/src/controllers/UserController.ts

```diff
@@ -155,4 +171,143 @@ async function getPermissions(user: Doc<UserModel>): Promise<Permission[]> {
         .map((permission) => Permission[permission as keyof typeof Permission]);
 }
 
-export { devLoginUser, loginUser, getAllUsers, updateUser, getPermissions };
+/**
+ * @desc 	Gives user roles specified in request body
+ * @route 	POST /api/users/:userId/roles/
+ */
+
+const giveRoles = asyncHandler(async (req: TypedRequest<{ roles: string[] }, UserIdParam>, res: Response) => {
+    // Check atleast one role specified
+    if (req.body.roles.length == 0) {
+        res.status(400).json({
+            status: 'error',
+            message: 'You must specify atleast one role',
+        });
+        return;
+    }
```

Comment:
Before this, you should check that the `roles` is actually an array using `Array.isArray(req.body.roles)`. Unfortunately, just because we type it in Typescript doesn't mean it does any actual checking that the types match during runtime.

Reply:
Just a note, we only need to do this in places where we're not passing them directly to mongoose as that automatically validates the types